src/responder/nss/nsssrv_mmap_cache.c

```diff
@@ -1445,8 +1446,15 @@ errno_t sss_mmap_cache_init(TALLOC_CTX *mem_ctx, const char *name,
     /* Attempt allocation several times, in case of EINTR */
     for (int i = 0; i < POSIX_FALLOCATE_ATTEMPTS; i++) {
         ret = posix_fallocate(mc_ctx->fd, 0, mc_ctx->mmap_size);
-        if (ret != EINTR)
+        if (ret == EINTR)
+            continue;
+        /* Copy-on-write file systems such as ZFS and Btrfs can't
+         * really support the posix_fallocate operation.
+         * Fall back to ftruncate() in this case */
+        if (ret == ENOSYS || ret == EOPNOTSUPP) {
+            ret = ftruncate(mc_ctx->fd, mc_ctx->mmap_size);
             break;
```

Comment:
Shouldn't this `break` be outside of the `if`-block?

If the call to `posix_allocate()` returns `EOK` or any other error, we will continue looping. This is the opposite of what was done before.

Reply:
Indeed.